Doc/library/stdtypes.rst

```diff
@@ -2573,8 +2573,9 @@ arbitrary binary data.
             bytearray.partition(sep)
 
    Split the sequence at the first occurrence of *sep*, and return a 3-tuple
-   containing the part before the separator, the separator, and the part
-   after the separator.  If the separator is not found, return a 3-tuple
+   containing the part before the separator, the separator itself or its
+   bytearray copy, and the part after the separator.
+   If the separator is not found, return a 3-tuple
```

Comment:
The change is to add 'itself or its bytearray copy'. The asymmetry surprised me, but it matches current behavior.
```
>>> b'123456789'.partition(bytearray(b'45'))
(b'123', bytearray(b'45'), b'6789')
>>> bytearray(b'123456789').partition(b'45')
(bytearray(b'123'), bytearray(b'45'), bytearray(b'6789'))
```
The text and test changes look good.